### PR TITLE
[Fix](nereids) forward the user define variables to master

### PR DESCRIPTION
**Pull request by feiniaofeiafei**

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -495,6 +495,11 @@ struct TFeResult {
     1001: optional bool noAuth
 }
 
+struct TTypeAndLiteral {
+    1: required Types.TPrimitiveType type
+    2: required string value
```
Comment (morrySnow) [inline gensrc/thrift/FrontendService.thrift:500]: do not use required, use optional

Comment (morrySnow) [inline gensrc/thrift/FrontendService.thrift:500]: why value is string? i think we should use TExprNode

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java [hunk 2]
```diff
@@ -479,4 +481,22 @@ public boolean isZero() {
...
+            case INT_LITERAL: return new IntLiteral(node.int_literal.value);
+            case LARGE_INT_LITERAL: return new LargeIntLiteral(node.large_int_literal.value);
+            case FLOAT_LITERAL: return new FloatLiteral(node.float_literal.value);
+            case DECIMAL_LITERAL: return new DecimalLiteral(node.decimal_literal.value);
+            case STRING_LITERAL: return new StringLiteral(node.string_literal.value);
+            case JSON_LITERAL: return new JsonLiteral(node.json_literal.value);
+            case DATE_LITERAL: return new DateLiteral(node.date_literal.value);
+            case IPV4_LITERAL: return new IPv4Literal(node.ipv4_literal.value);
+            case IPV6_LITERAL: return new IPv6Literal(node.ipv6_literal.value);
+            default: throw new AnalysisException("Wrong type from thrift;");
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java:499]: maybe we should skip deserialize this literal?

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 3]
```diff
@@ -654,4 +663,18 @@ public TMasterOpResult proxyExecute(TMasterOpRequest request) {
...
+        try {
+            Map<String, LiteralExpr> userVariables = Maps.newHashMap();
+            for (Map.Entry<String, TExprNode> entry : thriftMap.entrySet()) {
+                TExprNode tExprNode = entry.getValue();
+                LiteralExpr literalExpr = LiteralExpr.getLiteralExprFromThrift(tExprNode);
+                userVariables.put(entry.getKey(), literalExpr);
+            }
+            return userVariables;
+        } catch (AnalysisException e) {
+            throw new TException(e.getMessage());
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:677]: ditto

**Review by 924060929 — APPROVED** [review]

**Pull request merged**